scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/connectionpooling/DBConnectionPoolsHolder.java

```diff
@@ -0,0 +1,130 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.examples.connectionpooling;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import javax.sql.rowset.CachedRowSet;
+import javax.sql.rowset.RowSetProvider;
+
+import org.apache.log4j.Logger;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.zaxxer.hikari.HikariDataSource;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 28/12/2018
+ */
+public class DBConnectionPoolsHolder {
+
+    private static final int CACHE_MAXIMUM_SIZE = Integer.parseInt(System.getProperty("external.db.pool.cache.maximum.size",
+                                                                                      "10"));
+
+    private static final int CACHE_EXPIRATION_MINUTES = Integer.parseInt(System.getProperty("external.db.pool.cache.expiration.minutes",
+                                                                                            "10"));
+
+    private DBConnectionPoolsHolder() {
+    }
+
+    private final LoadingCache<DBConnectionDetails, HikariDataSource> dbConnectionPoolMap = CacheBuilder.newBuilder()
+                                                                                                        .maximumSize(CACHE_MAXIMUM_SIZE)
+                                                                                                        .expireAfterAccess(CACHE_EXPIRATION_MINUTES,
+                                                                                                                           TimeUnit.MINUTES)
+                                                                                                        .build(new CacheLoader<DBConnectionDetails, HikariDataSource>() {
+                                                                                                            public HikariDataSource
+                                                                                                                    load(DBConnectionDetails key)
+                                                                                                                            throws SQLException {
+                                                                                                                System.out.println("New connection to an external DB is created " +
```

Comment:
please replace System.out.println with a proper logger....maybe taskLogger?

Reply:
We cannot as this is executed by the Task. Any logger.info will be printed to the Node log file and not to the task output.

So if we want to see this in the task output there is currently no other way than to do System.out.println